docs/design.rst

```diff
@@ -1,29 +1,70 @@
 .. _schemaregistry_design:
 
 Design Overview
----------------
+===============
 The Schema Registry is a distributed storage layer for Avro Schemas which uses Kafka as its underlying storage mechanism. Some key design decisions:
 
-* Assigns globally unique id to each registered schema. Allocated ids are guaranteed to be monotonically increasing but not necessarily consecutive, and ZooKeeper is used to to help maintain this guarantee.
+* Assigns globally unique id to each registered schema. Allocated ids are guaranteed to be monotonically increasing but not necessarily consecutive.
 * Kafka provides the durable backend, and functions as a write-ahead changelog for the state of the Schema Registry and the schemas it contains.
-* The Schema Registry is designed to be distributed, with single-master architecture, and ZooKeeper coordinates master election.
+* The Schema Registry is designed to be distributed, with single-master architecture, and ZooKeeper/Kafka coordinates master election (based on the configuration).
 
-.. image:: schema-registry-design.png
+Schema ID Allocation
+--------------------
+
+SchemaId allocation always happen in the master node and they ensure that the Schema Ids are
```

Comment:
nit: i think we're just inconsistent in these docs wrt capitalization already, but I think we generally don't include the internal implementation class names. Should really be either Schema ID, Schema Id or Schema id. (I personally like Schema ID in prose, but whichever is most consistent works fine.